src/layers/recurrent.jl

```diff
@@ -211,21 +220,36 @@ function (lstm::LSTMCell)(x::AbstractMatrix, ps::Union{ComponentArray, NamedTupl
     return lstm((x, hidden_state, memory), ps, st)
 end
 
-function (lstm::LSTMCell)((x, hidden_state,
-                           memory)::Tuple{<:AbstractMatrix, <:AbstractMatrix,
-                                          <:AbstractMatrix},
-                          ps::Union{ComponentArray, NamedTuple}, st::NamedTuple)
+function (lstm::LSTMCell{true})((x, hidden_state,
+                                 memory)::Tuple{<:AbstractMatrix, <:AbstractMatrix,
+                                                <:AbstractMatrix},
+                                ps::Union{ComponentArray, NamedTuple}, st::NamedTuple)
```

Comment:
We can remove `::Union{ComponentArray, ...}` now from all the calls.

Reply:
Done